app/src/main/java/com/example/healthinspector/Fragments/ScanFlow/ScanFragment.java

```diff
@@ -81,11 +95,52 @@ public void run() {
+                                    //getting "nutriscore_grade"
+                                    String healthInspectorScore = response.getJSONObject("product").getString("nutriscore_grade");
+                                    //getting the list of ingredients
+                                    ArrayList<String> ingredients = new ArrayList<>(Arrays.asList(response.getJSONObject("product").getString("ingredients_text_en").split(" ,")));
+                                    ingredients.set(ingredients.size()-1, ingredients.get(ingredients.size()-1).replace(".", ""));
+                                    //getting the ingredients analysis
+                                    JSONArray ingredientsAnalysisJSON = response.getJSONObject("product").getJSONArray("ingredients_analysis_tags");
+                                    ArrayList<String> ingredientsAnalysis = new ArrayList<>();
+                                    //convert JSONarray to String arraylist
+                                    for (int i = 0; i < ingredientsAnalysisJSON.length(); i++){
+                                        ingredientsAnalysis.add(ingredientsAnalysisJSON.getString(i));
+                                    }
+                                    //parsing the string to remove the "en:" prefix
+                                    for(int i = 0; i < ingredientsAnalysis.size(); i++){
+                                        String analysis = "";
+                                        for(int x = 3; x < ingredientsAnalysis.get(i).length(); x++){
+                                            analysis += ingredientsAnalysis.get(i).charAt(x);
+                                        }
+                                        ingredientsAnalysis.set(i, analysis);
+                                    }
+                                    //getting novaGroup of product
+                                    int novaGroupNumber = response.getJSONObject("product").getInt("nova_group");
+                                    String novaGroup = novaGroups.get(novaGroupNumber);
+                                    //  Log.i(TAG, novaGroup);
+
+                                    //getting "nutrient_levels" for warning
+                                    ArrayList<String> nutrientLevels = new ArrayList<>();
+                                    JSONObject nutrientLevelsJSON = response.getJSONObject("product").getJSONObject("nutrient_levels");
+                                    Iterator iterator =  nutrientLevelsJSON.keys();
+                                    while (iterator.hasNext()){
+                                        nutrientLevels.add(nutrientLevelsJSON.getString(iterator.next().toString()));
+                                    }
+                                    //send to next screen and populate views
+                                    //creating a scannedProduct object for to wrap with
```

Comment:
please get rid of theses comments. Comments in code should really only be for explaining something really complex or unintuitive.